app/validators/routes_and_redirects_validator.rb

```diff
@@ -21,7 +27,7 @@ def validate(record)
       redirects_must_not_have_routes(record, routes)
       redirects_must_include_base_path(record, redirects)
     else
-      routes_must_include_base_path(record, routes)
+      routes_must_include_base_path(record, routes) unless is_unpublishing
```

Comment:
I'd be in favour of not adding to those models where possible. Very easy for them to have lots of logic concerns that they don't need and they are used in many contexts.

So if we were to have just one class that did the validation both, I think it'd be the responsibility of that class to know how to extract the information needed - since it likely won't be needed in the same way in a different context. 

The other option for a single class would be for the validator to be called in a lambda and the key information injected in from that, which would at least encapsulate the validation concerns to the lambda. Bummer there would be if it requires a reasonable amount of code.